### PR TITLE
Fixes rhai entities

### PR DESCRIPTION
**Pull request by ldjebran**
test output https://github.com/SatelliteQE/robottelo/pull/6813

### CODE REVIEW

File: airgun/views/rhai.py [hunk 1]
```diff
@@ -35,9 +58,22 @@ def is_displayed(self):
         return self.title.is_displayed and self.search.is_displayed
 
 
+class InventoryHostRule(GenericRemovableWidgetItem):
+    """Insights inventory host rule widget"""
+    # There is no remove button in this widget
```
Comment (oshtaier) [inline airgun/views/rhai.py]: sounds weird. `GenericRemovableWidgetItem` and `There is no remove button in this widget`. `Removable`, but `no remove button`. That is not logical. Any alternative

File: airgun/entities/rhai/base.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from airgun.exceptions import InsightsOrganizationPageError
+from airgun.navigation import NavigateStep
+from airgun.views.rhai import InsightsOrganizationErrorView
+
+
+class InsightsNavigateStep(NavigateStep):
```
Comment (oshtaier) [inline airgun/entities/rhai/base.py]: propose to not use that navigation at all

**Comment by ldjebran** [comment]
output to latest changes https://github.com/SatelliteQE/robottelo/pull/6813#issuecomment-473348979

File: airgun/views/rhai.py [hunk 3]
```diff
@@ -10,7 +10,30 @@
...
+    title = Text("//article[@id='content']/section/h1")
+    message = Text("//article[@id='content']/section/p")
+
+    @property
+    def is_displayed(self):
+        return (self.browser.wait_for_element(self.title, exception=False) is not None
+                and self.browser.wait_for_element(self.message, exception=False) is not None)
+
+    def read(self, widget_names=None):
+        if widget_names:
```
Comment (oshtaier) [inline airgun/views/rhai.py]: as far as I understand, we know that `if not self.view.is_displayed:`  and we know that no widgets are displayed, so I suggest to use plain:
`return '{0}: {1}'.format(self.title.read(), self.message.read())`

File: airgun/views/rhai.py (hunk 1 above)
Comment (oshtaier) [reply airgun/views/rhai.py]: Maybe we can create an issue per our discussion? To create better architecture for such kind of widgets?

**Comment by ldjebran** [comment]
@oshtaier changes done and issue https://github.com/SatelliteQE/airgun/issues/305 created

File: airgun/views/rhai.py [hunk 4]
```diff
@@ -10,7 +10,28 @@
...
+    title = Text("//article[@id='content']/section/h1")
+    message = Text("//article[@id='content']/section/p")
+
+    @property
+    def is_displayed(self):
+        return (self.browser.wait_for_element(self.title, exception=False) is not None
+                and self.browser.wait_for_element(self.message, exception=False) is not None)
+
+    def read(self, widget_names=None):
+        return '{0}: {1}'.format(self.title.read(), self.message.read())
```
Comment (abalakh) [inline airgun/views/rhai.py]: I'm not sure we should mess with custom read, especially as there's no real need to - both title and message will be returned correctly with default read implementation.

File: airgun/views/rhai.py [hunk 5]
```diff
@@ -10,7 +10,28 @@
...
+class InsightsOrganizationErrorView(BaseLoggedInView):
+    """View displayed when no Organization is selected or when the current organization has no
+    manifest
+    """
+    title = Text("//article[@id='content']/section/h1")
+    message = Text("//article[@id='content']/section/p")
+
+    @property
+    def is_displayed(self):
+        return (self.browser.wait_for_element(self.title, exception=False) is not None
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply airgun/views/rhai.py]: 1- This has a very specific usage in insight 
2- If you look at the locators 
title = Text("//article[@id='content']/section/h1")
message = Text("//article[@id='content']/section/p")
you will never fiver find a normal page with these 2 element at the same time for forman / kattelo there is not page like this even with only title, in the insight also there is no page with locator except this error page

File: airgun/views/rhai.py (hunk 4 above)
Comment (ldjebran) [reply airgun/views/rhai.py]: This really has no importance, is used in one place to get that error message, if we do not pack it here , we will pack else where, but really do not see, why we should have them separated, it is an error message.

**Pull request merged**